### PR TITLE
feat(isaaclab): add pi0.5 PPO finetuning example

### PR DESCRIPTION
**Pull request by YifWRobotics**
### Description

This PR adds support for finetuning **OpenPI π0.5 models with PPO in the IsaacLab environment** within the RLinf framework.

The main configuration for this workflow is provided in:

`examples/embodiment/config/isaaclab_franka_stack_cube_ppo_openpi_pi05.yaml`

This configuration enables **PPO-based finetuning and evaluation of π0.5 policies** for the Franka stack cube manipulation task in IsaacLab.

To support this workflow, the PR introduces several changes:

- Adds the π0.5 IsaacLab PPO configuration  
  `examples/embodiment/config/isaaclab_franka_stack_cube_ppo_openpi_pi05.yaml`

- Adds an IsaacLab-specific OpenPI data configuration  
  `rlinf/models/embodiment/openpi/dataconfig/isaaclab_dataconfig.py`

- Adds IsaacLab observation/action transforms for OpenPI policies  
  `rlinf/models/embodiment/openpi/policies/isaaclab_policy.py`

- Registers the π0.5 IsaacLab stack cube training configuration in  
  `rlinf/models/embodiment/openpi/dataconfig/__init__.py`

- Extends the OpenPI installation script to support the IsaacLab environment  
  `requirements/install.sh`

- Updates the Docker image to install both GR00T and OpenPI IsaacLab environments  
  `docker/Dockerfile`

The example demonstrates how to use PPO to finetune a π0.5 policy for robotic manipulation tasks using:

- RGB observations from front and wrist cameras  
- robot state observations  
- natural language task prompts

The task used in this example is the **Franka stack cube task**, where the robot stacks blocks in the order:

blue → red → green

The PR also provides reproducible command-line workflows for running the example in different environments:

- HPC clusters using Apptainer containers  
- local machines using RLinf dependency installation  
- Docker-based environments

These workflows include commands for:

- environment setup
- downloading the π0.5 SFT checkpoint
- running evaluation
- running PPO finetuning

### Motivation and Context

The current IsaacLab example in RLinf demonstrates finetuning **GR00T models** with PPO. This PR extends the workflow to also support **π0.5 models**.

By adding this example, users can understand how to apply RLinf to **finetune π0.5 policies using PPO in IsaacLab environments**. This makes it easier to reproduce and extend embodied learning experiments with π0.5 within the RLinf framework.

The added instructions also provide **practical deployment workflows** for both HPC clusters and local machines, which helps users reproduce the experiments in different computing environments.

### How has this been tested?

Testing was performed by running the IsaacLab stack cube training and evaluation pipelines using the π0.5 configuration.

The following checks were performed:

- environment initialization in IsaacLab
- launching PPO finetuning with the stack cube task
- verifying rollout collection and PPO optimization steps
- running the evaluation script successfully
- running the training script successfully

The example was tested in the following environments:

- **HPC cluster using Apptainer container**
- **Local PC installation using RLinf dependency scripts**
- **Docker container environment**

### Additional information (optional, e.g., figures and logs):

The example uses a π0.5 supervised-finetuned checkpoint trained for the IsaacLab Stack Cube Task.

HuggingFace checkpoint: [YifWRobotics/RLinf-pi05-SFT-Stack-cube](https://huggingface.co/YifWRobotics/RLinf-pi05-SFT-Stack-cube)

The checkpoint is a supervised-finetuned (SFT) π0.5 model used as the initialization policy before PPO finetuning in the IsaacLab environment. This SFT checkpoint was trained using the **IsaacLab Stack Cube demonstration dataset**:

Dataset:  [RLinf/IsaacLab-Stack-Cube-Data](https://huggingface.co/datasets/RLinf/IsaacLab-Stack-Cube-Data)

Prebuilt Docker image used during testing:

Docker tag:
https://hub.docker.com/repository/docker/yifwrobotics/rlinf/tags/isaaclab-openpi-gr00t

Pull command:
docker pull yifwrobotics/rlinf:isaaclab-openpi-gr00t

### Demo

Example PPO rollout of the π0.5 policy on the IsaacLab Franka Stack Cube task.

https://github.com/user-attachments/assets/bd90d71b-f62a-425b-b30d-bbf73eecd232

### Example Training Command

Example command used to launch PPO finetuning for the π0.5 policy on the IsaacLab Franka Stack Cube task:

```bash
python /workspace/RLinf/examples/embodiment/train_embodied_agent.py \
 --config-path /workspace/RLinf/examples/embodiment/config/ \
 --config-name isaaclab_franka_stack_cube_ppo_openpi_pi05 \
 runner.logger.log_path=/workspace/RLinf/logs/train_$(date +%Y%m%d-%H%M%S) \
 runner.logger.project_name=rlinf \
 runner.logger.experiment_name=isaaclab_ppo_openpi_pi05 \
 'runner.logger.logger_backends=[tensorboard,wandb]' \
 actor.micro_batch_size=8 \
 actor.global_batch_size=128 \
 runner.val_check_interval=20 \
 algorithm.update_epoch=3 \
 algorithm.rollout_epoch=1 \
 algorithm.kl_beta=0.01 \
 algorithm.entropy_bonus=0.001 \
 actor.optim.lr=2e-6 \
 actor.optim.value_lr=5e-5 \
 actor.fsdp_config.gradient_checkpointing=False \
 rollout.model.model_path=/workspace/RLinf/outputs/RLinf-pi05-SFT-Stack-cube \
 actor.model.model_path=/workspace/RLinf/outputs/RLinf-pi05-SFT-Stack-cube
```

After 66 steps, the success rate rises from 0.859 to 0.953:
Screenshot from 2026-03-17 15-55-08

In another run with lower initial success rate, after 13 steps, the training plot looks like
Screenshot from 2026-03-16 15-49-42

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -235,7 +235,8 @@ RUN echo "source ${UV_PATH}/openpi/bin/activate" >> ~/.bashrc
 FROM embodied-common-image AS embodied-isaaclab-image
 
 # Install gr00t env
-RUN bash requirements/install.sh embodied --venv gr00t --model gr00t --env isaaclab
+RUN bash requirements/install.sh embodied --venv gr00t --model gr00t --env isaaclab && \
+    bash requirements/install.sh embodied --venv openpi --model openpi --env isaaclab
```
Comment (XuS1994) [inline docker/Dockerfile:239]: Why add openpi env in `# Install gr00t env`

File: examples/embodiment/config/isaaclab_franka_stack_cube_ppo_openpi_pi05.yaml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    adam_beta1: 0.9
+    adam_beta2: 0.95
+    adam_eps: 1.0e-08
+    clip_grad: 1.0
+    weight_decay: 0.01
+    critic_warmup_steps: 0
+
+  fsdp_config:
+    strategy: "fsdp"
+    gradient_checkpointing: True
```
Comment (andylin-hao) [inline examples/embodiment/config/isaaclab_franka_stack_cube_ppo_openpi_pi05.yaml]: Does openpi support gradient checkpointing?

File: docs/source-en/rst_source/examples/embodied/isaaclab.rst [hunk 3]
```diff
@@ -212,18 +214,157 @@ Update the `model_path` in the configuration file to point to the directory wher
...
+Use Docker image for the experiment.
+
+.. code:: bash
+
+   docker run -it --rm --gpus all \
+      --shm-size 32g \
+      --network host \
+      --name rlinf \
+      -v .:/workspace/RLinf \
+      yifwrobotics/rlinf:isaaclab-openpi-gr00t
```
Comment (andylin-hao) [inline docs/source-en/rst_source/examples/embodied/isaaclab.rst]: Change this to rlinf/rlinf:agentic-rlinf0.2-isaaclab. I'll update the image later.

File: docs/source-en/rst_source/examples/embodied/isaaclab.rst [hunk 4]
```diff
@@ -212,18 +214,157 @@ Update the `model_path` in the configuration file to point to the directory wher
...
+      # For mainland China users, you can use the following for better download speed:
+      # docker.1ms.run/yifwrobotics/rlinf:isaaclab-openpi-gr00t
```
Comment (andylin-hao) [inline docs/source-en/rst_source/examples/embodied/isaaclab.rst]: Also, missing instructions to switch venv. Add the following

```
Please switch to the corresponding virtual environment via the built-in `switch_env` utility in the image:

.. code:: bash

   source switch_env openpi
```

File: docs/source-en/rst_source/examples/embodied/isaaclab.rst [hunk 5]
```diff
@@ -212,18 +214,157 @@ Update the `model_path` in the configuration file to point to the directory wher
...
+
+   bash examples/embodiment/run_embodiment.sh isaaclab_franka_stack_cube_ppo_openpi_pi05
+
+To evaluate openpi π0.5 in the IsaacLab environment, run:
+
+.. code:: bash
+
+   bash examples/embodiment/eval_embodiment.sh isaaclab_franka_stack_cube_ppo_openpi_pi05
+
+**4. Alternative Launch Command**
```
Comment (andylin-hao) [inline docs/source-en/rst_source/examples/embodied/isaaclab.rst]: Remove this alternative command. This is not the recommended way of RLinf and may confuse users.

File: docs/source-zh/rst_source/examples/embodied/isaaclab.rst [hunk 6]
```diff
@@ -53,7 +55,7 @@ IsaacLab 是一个高度可定制的仿真平台，允许用户创建自定义
 如需添加自定义任务，通常需要以下三步：
 
 1. **自定义 IsaacLab 环境**：可参考 `IsaacLab-Examples <https://isaac-sim.github.io/IsaacLab/v2.3.0/source/overview/environments.html>`__ 中的可用环境；自定义环境项目可参考 `IsaacLab-Quickstart <https://isaac-sim.github.io/IsaacLab/v2.3.0/source/overview/own-project/index.html>`__。
-2. **在 RLinf 中配置训练环境**：参考 ``rlinf/envs/isaaclab/tasks/stack_cube.py``，将自定义脚本放到 ``rlinf/envs/isaaclab/tasks``，并在 ``rlinf/envs/isaaclab/__init__.py`` 中添加相关代码。
+2. **在 RLinf 中配置训练环境**：参考 ``RLinf/rlinf/envs/isaaclab/tasks/stack_cube.py``，将自定义脚本放到 ``RLinf/rlinf/envs/isaaclab/tasks``，并在 ``RLinf/rlinf/envs/isaaclab/__init__.py`` 中添加相关代码。
```
Comment (andylin-hao) [inline docs/source-zh/rst_source/examples/embodied/isaaclab.rst]: Revert this

File: docs/source-zh/rst_source/examples/embodied/isaaclab.rst [hunk 7]
```diff
@@ -207,18 +210,157 @@ Isaac Sim 下载
...
+使用 Docker 镜像运行实验。
+
+.. code:: bash
+
+   docker run -it --rm --gpus all \
+      --shm-size 32g \
+      --network host \
+      --name rlinf \
+      -v .:/workspace/RLinf \
+      yifwrobotics/rlinf:isaaclab-openpi-gr00t
```
Comment (andylin-hao) [inline docs/source-zh/rst_source/examples/embodied/isaaclab.rst]: Same as above

File: docs/source-zh/rst_source/examples/embodied/isaaclab.rst [hunk 8]
```diff
@@ -207,18 +210,157 @@ Isaac Sim 下载
...
+
+   bash examples/embodiment/run_embodiment.sh isaaclab_franka_stack_cube_ppo_openpi_pi05
+
+在 IsaacLab 环境中评估 openpi π0.5：
+
+.. code:: bash
+
+   bash examples/embodiment/eval_embodiment.sh isaaclab_franka_stack_cube_ppo_openpi_pi05
+
+**4. 替代启动命令**
```
Comment (andylin-hao) [inline docs/source-zh/rst_source/examples/embodied/isaaclab.rst]: Same. Remove.

File: rlinf/models/embodiment/openpi/dataconfig/isaaclab_dataconfig.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+                        "observation/wrist_image": "observation.images.wrist",
+                        "observation/state": "observation.state",
+                        "actions": "action",
+                    }
+                )
+            ]
+        )
+
+        data_transforms = _transforms.Group(
+            inputs=[libero_policy.LiberoInputs(model_type=model_config.model_type)],
```
Comment (andylin-hao) [inline rlinf/models/embodiment/openpi/dataconfig/isaaclab_dataconfig.py]: Why not use the isaaclab_policy's new IsaacLabInputs?

**Comment by andylin-hao** [comment]
image
